File: Biblioteca C++/Biblioteca.cpp
/*
********************************************************************************
   
                         SEGUNDO TRABALHO DE P.O.O
________________________________________________________________________________       
                         
                   Luiz Henrique Cherri     NUSP: 5991829
               
             Para utilizar somente siga as instrucoes dos menus
              
               Qualquer duvida avise-me: luizcherri@gmail.com
________________________________________________________________________________

 OBS: Os codigos foram compilados no windowsXP utilizando o compilador da GNU
      DEV-C++ 4.9.9.2
 
********************************************************************************
*/

#include <iostream>
#include "Item.h"
#include "Livro.h"
#include "Revista.h"
#include "Publicacao.h"
#include "Midia.h"
#include "Cd.h"
#include "Dvd.h"
#include "Vector.h"
#include <string.h>

using namespace std;

int main () {
    Vector *v = new Vector();
    int end = true;
    int opc;
    int opc2;
    
    int x;
    char a[30];
    char *b;
    
    
    do {
        printf ("1 - Inserir\n");
        printf ("2 - Buscar \n");
        printf ("3 - Listar \n");
        printf ("0 - Sair   \n>> ");
        scanf ("%d",&opc) ;
        
        switch (opc) {
                case 1:
                    printf ("\nO que?:   \n");
                    printf ("1 - Livro   \n");
                    printf ("2 - Revista \n");
                    printf ("3 - CD      \n");
                    printf ("4 - DVD     \n>> ");
                    scanf ("%d",&opc2);
                    
                  
                    if (opc2 == 1){
                       Livro *r = new Livro();
                       
                       printf ("\nCodigo >> ");
                       scanf ("%d", &x);
                       r->setCod(x);
                       
                       printf ("Disponibilidade >> ");
                       fflush(stdin);
                       gets (a);
                       r->setDispo(a);
                       a[0] = '\0';
                       
                       printf ("Numero de paginas >> ");
                       scanf ("%d", &x);
                       r->setNpag(x);
                       
                       printf ("Autor >> ");
                       fflush(stdin);
                       gets (a);
                       r->setAutor(a);
                       a[0] = '\0';
                  
                       printf ("ISBN >> ");
                       fflush(stdin);
                       gets (a);
                       r->setIsbn(a);
                       a[0] = '\0';
                       
                       v->add(r);
                    }            
                    if (opc2 == 2) {
                       Revista *r = new Revista();
                       v->add(r);
                       
                       printf ("\nCodigo >> ");
                       scanf ("%d", &x);
                       r->setCod(x);
                       
                       printf ("Disponibilidade >> ");
                       fflush(stdin);
                       gets (a);
                       r->setDispo(a);
                       a[0] = '\0';
                       
                       printf ("Numero de paginas >> ");
                       scanf ("%d", &x);
                       r->setNpag(x);
                       
                       printf ("Editora >> ");
                       fflush(stdin);
                       gets (a);
                       r->setEditora(a);
                       a[0] = '\0';
                       
                       printf ("ISSN >> ");
                       fflush(stdin);
                       gets (a);
                       r->setIssn(a);
                       
                       a[0] = '\0';
                    }
                    if (opc2 == 3) {
                       Cd *r = new Cd();
                       
                       printf ("\nCodigo >> ");
                       scanf ("%d", &x);
                       r->setCod(x);
                       
                       printf ("Disponibilidade >> ");
                       fflush(stdin);
                       gets (a);
                       r->setDispo(a);
                       a[0] = '\0';
                       
                       printf ("Conteudo >> ");
                       fflush(stdin);
                       gets (a);
                       r->setConteudo(a);
                       a[0] = '\0';
                       
                       printf ("Dual Layer >> ");
                       fflush(stdin);
                       gets (a);
                       r->setDul(a);
                       
                       printf ("Duracao >> ");
                       fflush(stdin);
                       gets (a);
                       r->setDuration(a);
                       a[0] = '\0';

                       v->add(r);
                    }
                    if (opc2 == 4) {
                       Dvd *r = new Dvd();
                       
                       printf ("\nCodigo >> ");
                       scanf ("%d", &x);
                       r->setCod(x);
                       
                       printf ("Disponibilidade >> ");
                       fflush(stdin);
                       gets (a);
                       r->setDispo(a);
                       a[0] = '\0';
                       
                       printf ("Conteudo >> ");
                       fflush(stdin);
                       gets (a);
                       r->setConteudo(a);
                       a[0] = '\0';
                       
                       printf ("Dual Layer >> ");
                       fflush(stdin);
                       gets (a);
                       r->setDul(a);
                       
                       printf ("Numero de Trilhas >> ");
                       fflush(stdin);
                       gets (a);
                       r->setTrilhas(a);
                       a[0] = '\0';
                       
                       v->add(r);
                    }                                   
                               
                    break;
                    
                case 2: 
                     printf ("\n\nDigite o codigo que deseja buscar >> ");
                     scanf ("%d", &x);
                     for (int i = 0 ; i < v->size() ; i++) {
                         if (x == v->get(i)->getCodigo()) {
                              v->get(i)->imprimir();
                         }        
                     }
                     break;
                     
                case 3:
                     
                     for (int i = 0 ; i < v->size() ; i++) {
                         v->get(i)->imprimir();
                     }
                      
                     break;
                     
        }     

    }while (opc != 0);
    
    delete (v);
    
    getchar();    
}

File: src Biblioteca Java/Biblioteca.java
/**
 * SEGUNDO TRABALHO DE P.O.O.
 * @author Luiz Henrique Cherri
 * @NUSP 5991829
 *
 * Para utilizar, somente siga os menus...
 *
 * Feito no windowsXP, compilado no NetBeans IDE 6.5.1
 *
 * Qualquer problema avise-me: luizcherri@gmail.com
 */

import java.util.*;
import java.io.*;
import java.lang.String;
public class Biblioteca {


    public static void main(String[] args) {
        
        Vector<Item> v = new Vector<Item>();
        int opc;
        int itype;
        int x;
        String s;

        Scanner sc = new Scanner (System.in);
        do {
            System.out.println ("\n--- BIBLIOTECA ---");
            System.out.println ("1 - Inserir");
            System.out.println ("2 - Buscar ");
            System.out.println ("3 - Listar ");
            System.out.println ("0 - Sair   ");
            
            opc = sc.nextInt(); sc.nextLine();

            switch (opc){
                case 1 :
                    System.out.println ("\nInserir o que?");
                    System.out.println ("1 - Livro  ");
                    System.out.println ("2 - Revista");
                    System.out.println ("3 - CD     ");
                    System.out.println ("4 - DVD    ");
                    itype = sc.nextInt(); sc.nextLine();
                    switch (itype){
                        case 1 :
                            Livro l = new Livro();
                            System.out.print ("Codigo >> ");
                            x = sc.nextInt(); sc.nextLine();
                            l.setCodigo(x);

                            System.out.print ("Disponibilidade >> ");
                            s = sc.nextLine();
                            l.setDisp(s);

                            System.out.print ("Numero de pagina >> ");
                            x = sc.nextInt(); sc.nextLine();
                            l.setNpag(x);

                            System.out.print ("Autor >> ");
                            s = sc.nextLine();
                            l.setAutor(s);

                            System.out.print ("ISBN >> ");
                            s = sc.nextLine();
                            l.setIsbn(s);
                            v.add (l);
                            break;
                        case 2 :
                            Revista r = new Revista();
                            System.out.print ("Codigo >> ");
                            x = sc.nextInt(); sc.nextLine();
                            r.setCodigo(x);

                            System.out.print ("Disponibilidade >> ");
                            s = sc.nextLine();
                            r.setDisp(s);

                            System.out.print ("Numero de pagina >> ");
                            x = sc.nextInt(); sc.nextLine();
                            r.setNpag(x);

                            System.out.print ("Editora >> ");
                            s = sc.nextLine();
                            r.setEditora(s);

                            System.out.print ("ISSN >> ");
                            s = sc.nextLine();
                            r.setIssn(s);
                            v.add (r);
                            break;
                        case 3 :
                            CD c = new CD();
                            System.out.print ("Codigo >> ");
                            x = sc.nextInt(); sc.nextLine();
                            c.setCodigo(x);

                            System.out.print ("Disponibilidade >> ");
                            s = sc.nextLine();
                            c.setDisp(s);

                            System.out.print ("Conteudo >> ");
                            s = sc.nextLine();
                            c.setConteudo(s);

                            System.out.print ("Dual Layer >> ");
                            s = sc.nextLine();
                            c.setDul(s);

                            System.out.print ("Duracao >> ");
                            s = sc.nextLine();
                            c.setDuracao(s);




                            v.add (c);
                            break;
                        case 4 :
                            DVD d = new DVD();

                            System.out.print ("Codigo >> ");
                            x = sc.nextInt(); sc.nextLine();
                            d.setCodigo(x);

                            System.out.print ("Disponibilidade >> ");
                            s = sc.nextLine();
                            d.setDisp(s);

                            System.out.print ("Conteudo >> ");
                            s = sc.nextLine();
                            d.setConteudo(s);

                            System.out.print ("Dual Layer >> ");
                            s = sc.nextLine();
                            d.setDul(s);

                            System.out.print ("Numero de Fixas >> ");
                            x = sc.nextInt(); sc.nextLine();
                            d.setTracks(x);

                            v.add (d);
                            break;
                    }
                    break;

                case 2 : // BUSCAR
                    System.out.print ("Digite o codigo >> ");
                    x = sc.nextInt(); sc.nextLine();
                    for (int i = 0 ; i < v.size() ; i++){
                        if (x == v.elementAt(i).getCodigo()){
                            
                            v.elementAt(i).imprimir();
                        }

                    }
                    break;
                
                case 3 : // LISTAR
                    for (int i = 0 ; i < v.size() ; i++){
                        v.elementAt(i).imprimir();
                    }
                    break;

            }
        } while (opc != 0);
    }
}
